pkg/sql/plan/function/func_unary.go

```diff
@@ -1001,6 +1009,560 @@ func Unhex(parameters []*vector.Vector, result vector.FunctionResultWrapper, pro
 	return nil
 }
 
+//func createOllamaEmbedding(ctx context.Context, input string) ([]float32, error) {
+//	llm, err := ollama.New(ollama.WithModel("llama3"))
+//	if err != nil {
+//		return nil, err
+//	}
+//
+//	embeddings, err := llm.CreateEmbedding(ctx, []string{input})
+//	if err != nil {
+//		return nil, err
+//	}
+//
+//	return embeddings[0], nil
+//}
+
+type OllamaSingleEmbeddingRequest struct {
+	Model string `json:"model"`
+	Input string `json:"input"`
+}
+
+type OllamaMultipleEmbeddingRequest struct {
+	Model string   `json:"model"`
+	Input []string `json:"input"`
+}
+
+type OllamaEmbeddingResponse struct {
+	Model           string      `json:"model"`
+	Embeddings      [][]float32 `json:"embeddings"`
+	TotalDuration   int64       `json:"total_duration"`
+	LoadDuration    int64       `json:"load_duration"`
+	PromptEvalCount int         `json:"prompt_eval_count"`
+}
+
+// Prepare & send the HTTP request, read the response body, return embeddings
```

Comment:
try to group ollama specified code in one file and create an interface on top of ollama so that other LLM model can share the same interface.  Explore LangChain interface.